File: Solutions by Programming Task/Fibonacci sequence/Java/Iterative.java
import java.util.stream.IntStream;

public class Main {
    private static int Fib(int x) {
        var p = 0;
        var n = 1;
        for (var __ : IntStream.range(0, x).toArray()) {
            var sum = p + n;
            p = n;
            n = sum;
        }
        return p;
    }

    public static void main(String[] args) {
        System.out.println(Fib(10));
    }
}

File: Solutions by Programming Task/Fibonacci sequence/Go/Iterative.go
package main

import "fmt"

func Fib(x int) int {
	d := [2]int{0,1}
	for range make([]int, x) {
		sum := d[0] + d[1]
		d[0] = d[1]
		d[1] = sum
	}
	return d[0]
}

func main() {
	fmt.Println(Fib(10))
}
